### DISCUSSION TITLE
Creating a representation of Curve448?

### DISCUSSION THREAD

**Discussion by gliese1337** [openssl/openssl#24019] [Q&A]
I want to use Curve448 for ECC with deterministic keys produced by PBKDF2, but it is not available as a named curve in OpenSSL. No problem, I should just be able to create a custom group as demoed in https://wiki.openssl.org/index.php/Elliptic_Curve_Cryptography . Except, EC_GROUP_new_curve_GFp seems to require parameters for a curve in Weierstrass form, and I can only find parameters for Curve448 in Montgomery or Edwards form. Is it possible to create a custom group for Curve448?

Alternatively, it looks like I can create a private X448 key with `EVP_PKEY_new_raw_private_key`--but if I do that, will it automatically fill in the public key?

**Comment by mattcaswell** [comment]
> Is it possible to create a custom group for Curve448?

No. You cannot use OpenSSL's `EC_*` functions for Curve448. OpenSSL uses an entirely separate key type for Ed448/X448:

https://www.openssl.org/docs/man3.2/man7/EVP_PKEY-ED448.html

> Alternatively, it looks like I can create a private X448 key with EVP_PKEY_new_raw_private_key--but if I do that, will it automatically fill in the public key?

Yes, it will. Of course you can also generate an entirely new X448 key using `EVP_PKEY_Q_keygen`:

https://www.openssl.org/docs/man3.2/man3/EVP_PKEY_Q_keygen.html

e.g.

```` C
    EVP_PKEY *pkey = EVP_PKEY_Q_keygen(NULL, NULL, "X25519");
````
